channels/active_channels.go

```diff
@@ -17,6 +17,8 @@ import (
 	"github.com/couchbase/sync_gateway/base"
 )
 
+type ChangedChannels map[ID]bool
```

Comment:
I think this might be a placeholder until user/roles are collection aware, because then the `TimedSet` returned from auth package are aware.

Reply:
To step back a bit, `ActiveChannels` is used to track replications for a database. It's basically a fancy counter, so we could use serialized formats, but at a potential cost. The question is whether we want to actually know active channels per replication by channel, or we are happy with a database level value.

Ideas:

* have ChangedChannels be a `ChangedKeys` again, because the keys could be a principal that doesn't have a channel. I could have converted everything to string and use the existing datastructures. `ActiveChannels` is really just a fancy counter.
* Otherwise we could use `ChangedKeys` per collection if we want to track channels per collection on a replication.
* We could create a datastructure that is actually `ActiveChannels` but it is collection scoped.